test/cli/gateway_test.go

```diff
@@ -129,7 +129,7 @@ func TestGateway(t *testing.T) {
 		t.Run("GET IPFS invalid CID returns 400 (Bad Request)", func(t *testing.T) {
 			t.Parallel()
 			resp := client.Get("/ipfs/QmInvalid/pleaseDontAddMe")
-			assert.Equal(t, 400, resp.StatusCode)
+			assert.Equal(t, 500, resp.StatusCode)
```

Comment:
I think this one needs to be restored to 400 – on `/ipfs/cid` we can always tell when CID is invalid, and that means we always know when it was 400 Bad Request, and not server error.

Reply:
Yes, requires https://github.com/ipfs/go-path/pull/66.

However, (https://github.com/ipfs/kubo/pull/9660#discussion_r1115178233) will still fail with 400 then. We probably should just create a follow-up issue for this to ensure name resolution errors bubble up as 500 instead. I think InvalidPath makes sense to be 400. Perhaps `path.IsValid` should not fail with hypothetical DNSLink domains? But that could break a lot of existent code.